### DISCUSSION TITLE
Returning custom error messages

### DISCUSSION THREAD

**Discussion by muety** [pocketbase/pocketbase#6102] [Q&A] [answered]
Using the JavaScript hooks API, how can I send a custom error message in the response? Some example code:

```javascript
onRecordCreate((e) => {
    throw new BadRequestError('Just a test', {'a': 'b'})
    e.next()
}, 'item')
```

The response body, however, is just this:

```
{"data":{},"message":"Failed to create record.","status":400}
```

I'm running PocketBase via `./pocketbase serve --dev`.

**Comment by ganigeorgiev** [comment]
Coming from https://github.com/pocketbase/pocketbase/discussions/3376, note that `onRecordCreate` is a record model hook and it differs from the related `onRecordCreateRequest` [request hook](https://pocketbase.io/docs/js-event-hooks/#request-hooks) (throwing an API error from the latter should still work).

Support for throwing API errors from non-request hooks was added in v0.23+ but in this case it seems like a bug caused by the JS exception -> Go error unwrapping. I'll have a more detailed look at it a little bit later. As a temp workaround you could try replacing `throw` with `return`.

**Reply by muety** [reply]
Thanks for clarifying the confusion about record- and request hooks. I meant to use a `onRecordCreateRequest` anyway. In this case, passing on the custom error works just fine. :+1:

**Reply by ganigeorgiev** [reply, marked as answer]
I've pushed a fix for the exception unwrapping in [v0.23.7 release](https://github.com/pocketbase/pocketbase/releases/tag/v0.23.7).

**Comment by vesamet** [comment]
Thank you for the fix! It saved my day ! ❤️
